### ISSUE TITLE
trackRequestResult: Skipped requests should mimic real request

### ISSUE THREAD

**Issue by oschlegel** [ngneat/elf#443]
### Which @ngneat/elf-* package(s) are relevant/releated to the feature request?

requests-status

### Description

When using `trackRequestResult` an empty observable is returned in case the request is cached, this is a problem when you want to subscribe to the resulting obserable in order to use the data from the request.

### Proposed solution

On success the request data should be stored in the request result. Instead of returning an empty obserable when skipping a request an obserable which emits the data from the request result should be returned.

### Alternatives considered

-

### Do you want to create a pull request?

Yes

**Comment by NetanelBasal** [comment]
We can do it, but why not read it from the store?

**Comment by oschlegel** [comment]
In some cases it's difficult to write a selector for the exact result of a request. For example if i have a requests which takes coordinates and a radius as parameters and returns a list of matching addresses. If i now execute the request multiple times with different parameters the store would contain addresses from all requests, but it wouldn't be possible anymore to tell which request returned which set of addresses

**Comment by NetanelBasal** [comment]
But the key should be based on the request params in this case

**Comment by oschlegel** [comment]
Yes, that's my current solution, but that looks like a workaround to me. I have to manipulate the request data and add a custom id, or add the request parameters as properties to each returned entity

**Comment by NetanelBasal** [comment]
Can you share a minimal example of the issue, please?

**Comment by NetanelBasal** [comment]
And what bothers you with that implementation?

**Comment by oschlegel** [comment]
The address returned by the request already has a unique id which i have to override. I would like to avoid to manipulate request data in general

**Comment by oschlegel** [comment]
If we would do the suggested change the same service could look like this:
```
interface Address {
  id: string;
  name: string;
}

@Injectable({ providedIn: 'root' })
export class AddressService {
  store = createStore({ name: 'address' }, withEntities<Address>());

  constructor(private httpClient: HttpClient) {}

  search(lat: number, lng: number, radius: number) {
    const query = `lng=${lng}&lat=${lat}&radius=${radius}`;
    return this.httpClient.get<Address[]>(`/api/address?${query}`).pipe(
      tap(addresses => this.store.update(addEntities(addresses))),
      trackRequestResult(['address', query])
    );
  }
}
```

**Comment by NetanelBasal** [comment]
You can create a PR that adds this option but I think it needs to be optional and not turn on by default

**Comment by oschlegel** [comment]
Do you think it could break applications if this would be the default behavior? I don't really see how since skipped requests and real requests would behave identically

**Comment by NetanelBasal** [comment]
It's not about breaking. It's about keeping it in memory when you don't need it.

**Comment by oschlegel** [comment]
ok, understand

**Comment by PraveenRL** [comment]
`trackRequestResult(['address', { skipCache: true }])` would help
